src/views/tickets/Yara.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { Segment, List } from 'semantic-ui-react';
+import { getArtifactListing, getArtifact } from '../../utils/api.js';
+
+const REFRESH_EVERY_MS = 1000;
+
+export default class Yara extends Component {
+  constructor(props) {
+    super(props);
+    console.log(props);
+
+    this.state = {
+      processed: false,
+      error: false,
+      isMalicous: false,
+      matchedRules: [],
+    };
+    this.processed = false;
+
+    this.refreshInterval = setInterval(
+      this.refreshTicketState,
+      REFRESH_EVERY_MS
+    );
+
+    this.refreshYaraState();
+  }
+
+  stopAutomaticRefreshing() {
+    clearInterval(this.refreshInterval);
+  }
+
+  async fetchArtifacts() {
+    const ticketId = this.props.ticketID;
+
+    const res = await getArtifactListing(ticketId);
+    const artifacts = res.fileArtifacts.filter(artifact =>
+      artifact.filename.endsWith('.yara')
+    );
+
+    let yara = [];
+
+    let i = 0;
+    for (i; i < artifacts.length; i++) {
+      let artifact = await getArtifact(
+        this.props.ticketID,
+        artifacts[i].filename
+      );
+      yara.push(artifact);
+    }
+    return yara;
+  }
+
+  refreshYaraState = async e => {
+    var yara = this;
```

Comment:
No `var`, use `.bind` or an arrow function to avoid the need to duplicate the scope.